staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go

```diff
@@ -208,6 +210,20 @@ type ValidationRule struct {
 	// "x must be less than max ("+string(self.max)+")"
 	// +optional
 	MessageExpression string
+	// reason represents a machine-readable description of why this validation failed.
+	// If this is the first validation in the list to fail, this reason, as well as the
```

Comment:
How does this work in combination with declarative validation failures?

Don't these validations produce field errors, not request-level errors? fieldPath makes sense, but I thought the types would map to some subset of field error types, not status types

https://github.com/kubernetes/kubernetes/blob/8233635d520d7c3fa7ea4764b559a9f1a40e2b5b/staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go#L105-L140

Reply:
Yes, will update